File: NapakalakiRuby/lib/monster.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

class Monster
  def initialize(n, l, b, p)
    @name = n
    @combatLevel = l
    @badConsequence = b
    @prize = p
  end
  
  def getName
    return @name
  end
  
  def getCombatLevel
    return @combatLevel
  end
  
  def getBadConsequence
    return @badConsequence
  end
  
  def getLevelsGained
    return @prize.getLevels
  end
  
  def getTreasuresGained
    return @prize.getTreasures
  end
  
  #NO UML
  
  def to_s
    cadena = "Name: #{@name} \nCombat_Level: #{@combatLevel}"
    cadena += "\nPrize: #{@prize} \nBad_Consequence: #{@badConsequence}"
    cadena += "\n\n"
    return cadena
  end
  
end
end

File: NapakalakiJava/src/napakalaki/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.*;

/**
 *
 * @author danibolanos
 */

public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badConsequence;
    
    public Monster(String n, int l, BadConsequence b, Prize p){
        name = n;
        combatLevel = l;
        badConsequence = b;
        prize = p;
    }
    
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public BadConsequence getBadConsequence(){
        return badConsequence;
    }
    
    public int getLevelsGained(){
        return prize.getLevels();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
    }
    
    //NO UML
    
    public Prize getPrize(){
        return prize;
    }
    
    public String toString(){
        return "Name : " + name + "\nCombat_Level : " + Integer.toString(combatLevel) 
        + "\nPrize : " + prize.toString() + "\nBad_Consequence : " 
        + badConsequence.toString();
    }
}
